mm/kfence/kfence.h

```diff
@@ -17,8 +17,8 @@
 #define PTR_FMT "%p"
 #endif
 
-/* Helper to get the canary byte pattern for @addr. */
-#define KFENCE_CANARY_PATTERN(addr) (((u8[]){ 0xaa, 0xab, 0xaa, 0xad })[(size_t)addr % 4])
+/* Get the canary byte pattern for @addr. */
+#define KFENCE_CANARY_PATTERN(addr) ((u8)0xaa ^ (u8)((unsigned long)addr & 0x7))
```

Comment:
Let's add some comment describing why we're using varying bytes.
E.g. "Use a rolling 8-byte pattern to detect memory corruptions with higher probability."

Reply:
Done.